packages/next/src/server-build.ts

```diff
@@ -2412,7 +2416,11 @@ export async function serverBuild({
           ]
         : [
             {
-              src: path.posix.join('/', entryDirectory, '.*'),
+              src: path.posix.join(
+                '/',
+                entryDirectory,
+                `${entryDirectory !== '.' ? '?' : ''}.*`
```

Comment:
fairly tricky logic for anyone unfamiliar with what an `entryDirectory` of `.` represents -- maybe can add a comment above about what this is handling